python/pyspark/ml/linalg/__init__.py

```diff
@@ -706,12 +706,12 @@ def dot(self, other: Iterable[float]) -> np.float64:
 
         elif isinstance(other, SparseVector):
             # Find out common indices.
-            self_cmind = np.in1d(self.indices, other.indices, assume_unique=True)
```

Comment:
according to https://numpy.org/doc/stable/reference/generated/numpy.in1d.html

> Deprecated since version 2.0: Use [isin](https://numpy.org/doc/stable/reference/generated/numpy.isin.html#numpy.isin) instead of [in1d](https://numpy.org/doc/stable/reference/generated/numpy.in1d.html#numpy.in1d) for new code.

Reply:
to fix:
```
starting mypy annotations test...
annotations failed mypy checks:
python/pyspark/ml/linalg/__init__.py:709: error: Module has no attribute "in1d"  [attr-defined]
python/pyspark/ml/linalg/__init__.py:714: error: Module has no attribute "in1d"  [attr-defined]
python/pyspark/mllib/linalg/__init__.py:816: error: Module has no attribute "in1d"  [attr-defined]
python/pyspark/mllib/linalg/__init__.py:821: error: Module has no attribute "in1d"  [attr-defined]
Found 4 errors in 2 files (checked 1044 source files)
1
```